### PR TITLE
Fix random seed for python/pytorch

### PR DESCRIPTION
**Pull request by whr94621**
fix random seed for random.shuffle;
fix random seed for initializaiton of pytorch;
add shuffle option to TextDataset.

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -17,6 +17,10 @@
 from src.utils.optim import Optimizer
 from src.utils.lr_scheduler import LossScheduler, NoamScheduler
 
+# Fix random seed
+torch.manual_seed(GlobalNames.SEED)
+if torch.cuda.is_available():
+    torch.cuda.manual_seed_all(GlobalNames.SEED)
```
Comment (zhengzx-nlp) [inline src/main.py]: 如果在 cpu 上就不 manual seed 吗？

Comment (whr94621) [reply src/main.py]: 这边cpu和gpu都要设立种子
